azure-pipelines.yml

```diff
@@ -0,0 +1,209 @@
+    displayName: Test with dotnet test
+    inputs:
+      command: test
+      projects: |
+        tests/**/*[Tt]ests/*.csproj
+        !tests/ApiPort/ApiPortVS.Tests/ApiPortVS.Tests.csproj
+        !**\PortabilityServiceIntegrationTests.dll
+      arguments: --configuration $(BuildConfiguration) --no-build
+
+  - task: CopyFiles@2
+    displayName: Copy Analysis files to Staging
+    inputs:
+      SourceFolder: bin\$(BuildConfiguration)
+      Contents: |
+        ApiPort\**\*
+        ApiPort.Offline\**\*
+        ApiPort.Vsix\*.vsix
+        raw\**\*
+        *.binlog
+      TargetFolder: $(Build.StagingDirectory)\drop
+
+  - task: CopyFiles@2
+    displayName: Collection symbol files
+    inputs:
+      SourceFolder: bin\$(BuildConfiguration)
+      Contents: '**\*.pdb'
+      TargetFolder: $(Build.StagingDirectory)\symbols
+
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-roslynanalyzers.RoslynAnalyzers@2
+    displayName: Run Roslyn Analyzers
+    inputs:
+      suppressionFileForCompilerWarnings: bin\RoslynAnalyzerSuppressions.txt
+
+  - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
+    displayName: Component Detection
+
+  - publish: $(Build.StagingDirectory)\drop
+    displayName: Publish drop
+    artifact: drop
+    enabled: false
+
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-fxcop.FxCop@2
+    displayName: 'Run FxCop'
+    inputs:
+      inputType: Basic
+      targets: 'bin\$(BuildConfiguration)\ApiPort*\net4*\**\ApiPort*.*;bin\$(BuildConfiguration)\ApiPort.Offline\net4*\**\Microsoft.Fx.*.dll;'
+      recursive: false
+      directory: 'C:\Windows\Microsoft.NET\Framework\v4.0.30319'
+      verbose: true
+    enabled: false
+    continueOnError: true
+
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-binskim.BinSkim@3
+    displayName: 'Run BinSkim '
+    inputs:
+      InputType: Basic
+      AnalyzeTarget: '$(Build.StagingDirectory)\drop\Microsoft.Fx.*.dll;$(Build.StagingDirectory)\drop\ApiPor*.dll;$(Build.StagingDirectory)\drop\ApiPor*.exe'
+      AnalyzeSymPath: $(Build.StagingDirectory)\symbols
+    continueOnError: true
+
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-apiscan.APIScan@1
+    displayName: 'Run APIScan net461'
+    inputs:
+      softwareFolder: '$(Build.StagingDirectory)\drop\raw\binaries\net461'
+      softwareName: 'API Port'
+      softwareVersionNum: 4.XX
+      symbolsFolder: '$(Build.StagingDirectory)\drop\raw\symbols\net461'
+      isLargeApp: false
+
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-publishsecurityanalysislogs.PublishSecurityAnalysisLogs@2
+    displayName: 'Publish Security Analysis Logs'
+    condition: succeededOrFailed()
+
+- job: Linux
+  pool:
+    vmImage: Ubuntu 18.04
+  variables:
+    BuildNetCoreAppTarget: netcoreapp2.1
+    RestoreBuildProjects: src/ApiPort/ApiPort/*.csproj
```

Comment:
Do we need to restore the lib projects under src\lib\**\*.csproj here?

Reply:
I looked into current dotnet-ApiPort-Linux-CI definition, It only has "src/ApiPort/ApiPort/*.csproj", I guess that you are just doing what it was doing here. I haven't worked on Linux build definition before, I'm not sure if it was broken in the UI definition always. Anyway, we can keep it as is, and test it out later.

Reply:
Yup: there are oh-so-many improvements/fixes I'd like to make, but I'm holding off so that we can review each of those changes individually in PRs. If I made a bunch of changes here as part of this PR, it would be hard to know what I *changed* vs. what I reproduced from your existing pipelines.